jmdict.go

```diff
@@ -109,10 +120,11 @@ func createSearchTerm(headword headword, entry jmdict.JmdictEntry, meta jmdictMe
 		Expression: headword.Expression,
 		Sequence:   -entry.Sequence,
 	}
-	for _, sense := range entry.Sense {
-		rules := grammarRules(sense.PartsOfSpeech)
-		term.addRules(rules...)
-	}
+
+	partsOfSpeech := meta.seqToPartsOfSpeech[entry.Sequence]
```

Comment:
is `meta.seqToPartsOfSpeech` guaranteed to contain `entry.Sequence` here?

Reply:
Yes, it's assigned in the "AddEntry" function, which is called on every entry (sequence) in the dictionary file regardless of language. If an entry doesn't contain any part-of-speech tags, it will be assigned an empty array (although I don't believe this ever happens in practice).

https://github.com/stephenmk/yomichan-import/blob/a9d85dc720e7b46c90f1efa1005b23e517415aa0/jmdict_metadata.go#L153-L167

![addentry](https://user-images.githubusercontent.com/8003332/216753204-d75b6951-0047-422d-af6c-ada9fd7ec4ad.png)